### PR TITLE
Http client to remove boilerplate unwrapping for requests

### LINKED ISSUE #308: Wrap http.get function to remove boiler plate code
We should write a custom implementation of get() with the HTTP class that wraps the requests.get() function to remove duplicate code like this across the codebase
```
    response = http.get(f"{badger_api}/tokens?chain={chain}")
    if not response.ok:
        return
    return response.json()
```

Something like
```
def fetch_data(link: str) -> Optional[Dict]:
    response = http.get(link)
    if not response.ok:
         log in discord
         return
    return response.json()
```

### PR DESCRIPTION
**Pull request by SHAKOTN**
**Problem:**
Wrap http.get function to remove boiler plate code

**Done:**
- Add `HTTPClient` object that acts as `requests` library proxy by doing `_unwrap_response` on each HTTP method
- Use `http_client` now in all internal requests instead of `http` session object
- Remove boilerplate code
- Move `mock_discord` to `conftest.py` to be accessible for all tests
- Tests

Fixes #308

### CODE REVIEW

File: helpers/http_client.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class HTTPClient:
+    """
+    Wrapper class on top of requests library
+    Serves as unwrapper on each HTTP request to avoid boilerplate request.status_code checking
+
+    All method signatures are matching respective requests.HTTP_method()
+
+    For example, self.get(self, url, **kwargs) is mathching requests.get(self, url, **kwargs)
```
Comment (mitche50) [inline helpers/http_client.py:16]: nit, but typo here

File: helpers/http_client.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    Implement more HTTP methods as needed
+    """
+    def __init__(self, http_session: requests.Session):
+        self.http = http_session
+        assert type(self.http) == requests.Session
+
+    @staticmethod
+    def _unwrap_response(response_obj: Response) -> Optional[Union[Dict, List]]:
+        if response_obj.ok:
+            return response_obj.json()
```
Comment (mitche50) [inline helpers/http_client.py:28]: What happens here if response_obj.ok is not True?  Should there be an explicit return None or is that implicit?

Comment (SHAKOTN) [reply helpers/http_client.py:28]: If there is no explicit return in the function, None will be returned by default.

File: helpers/http_client.py (hunk 1 above)
Comment (SHAKOTN) [reply helpers/http_client.py]: Fixed

**Review by Darruma — APPROVED** [review]

**Review by mitche50 — APPROVED** [review]

**Review by mitche50 — APPROVED** [review]

**Pull request merged**